rir/src/utils/FunctionWriter.h

```diff
@@ -164,44 +130,27 @@ class FunctionHandle {
         assert(skiplistLength == 0);
         assert(compressed + sources_idx == instruction_number);
 
-        code.code->srcLength -= compressed;
+        code->srcLength -= compressed;
 
         function->size -= compressed * sizeof(unsigned);
 
-        assert(code.code->srcLength == sources_idx);
+        assert(code->srcLength == sources_idx);
         function->codeLength++;
 
         // set the last code offset
         function->foffset = offset;
 
-        memcpy(callSites(code.code), callSiteBuffer, callSiteLength);
+        memcpy((void*)code->callSites(), callSiteBuffer, callSiteLength);
```

Comment:
Also C++ cast? Probably a lot of other places as well, maybe we should leave that to a separate cleanup PR?

Reply:
yeah,  idon't like them. I usually do not write reinterpret_cast, since it is essentially the same as a C style cast, just a more ugly name